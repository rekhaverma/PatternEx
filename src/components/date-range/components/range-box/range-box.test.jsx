import React from 'react';
import moment from 'moment';
import RangeBox from './range-box.component';

/**
 * @status: Complete
 * @sign-off-by: Alex Andries
 */
describe('<RangeBox />', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      className: 'range-box',
      currentDate: moment.utc('2018-01-01'),
      endDate: moment.utc('2018-01-01'),
      startDate: moment.utc('2018-01-01'),
      onDatesChange: jest.fn(),
      onMonthChange: jest.fn(),
    };
    wrapper = shallow(<RangeBox {...props} />);
  });

  it('should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the component with .range-box', () => {
    expect(wrapper.find('.range-box__flex').length).toEqual(1);
  });

  it('should trigger onDatesChange when method onDatesChange is called', () => {
    wrapper.instance().onDatesChange(moment.utc('2018-01-01'));

    expect(props.onDatesChange.mock.calls.length).toBe(1);
  });

  it('should trigger onDatesChange when method onSelectHour is called', () => {
    wrapper.instance().onSelectHour('12:00');

    expect(props.onDatesChange.mock.calls.length).toBe(1);
  });

  it('should trigger onMonthChange when method onMonthChange is called', () => {
    wrapper.instance().onMonthChange(moment.utc('2018-01-01'));

    expect(props.onMonthChange.mock.calls.length).toBe(1);
  });
});
