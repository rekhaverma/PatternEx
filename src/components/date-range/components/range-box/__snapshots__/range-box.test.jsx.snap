// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RangeBox /> should match with snapshot 1`] = `
<div
  className="range-box__flex"
>
  <div>
    <div
      className="range-box__inputs"
    >
      <div
        className="range-box__group"
      >
        <span>
          Start Date
        </span>
        <Input
          autoFocus={false}
          className="range-box__input"
          disabled={false}
          id=""
          name=""
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onKeyPress={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onSubmit={[Function]}
          placeholder=""
          style={Object {}}
          type="text"
          value="01-01-2018"
        />
      </div>
      <div
        className="icon-arrow-right"
      />
      <div
        className="range-box__group +right"
      >
        <span>
          End Date
        </span>
        <Input
          autoFocus={false}
          className="range-box__input +right"
          disabled={false}
          id=""
          name=""
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onKeyPress={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onSubmit={[Function]}
          placeholder=""
          style={Object {}}
          type="text"
          value="01-01-2018"
        />
      </div>
    </div>
    <DayPickerRangeController
      daySize={39}
      enableOutsideDays={false}
      endDate={"2018-01-01T00:00:00.000Z"}
      firstDayOfWeek={null}
      focusedInput="startDate"
      hideKeyboardShortcutsPanel={true}
      initialVisibleMonth={null}
      isDayBlocked={[Function]}
      isDayHighlighted={[Function]}
      isFocused={false}
      isOutsideRange={[Function]}
      isRTL={false}
      keepOpenOnDateSelect={false}
      minimumNights={0}
      monthFormat="MMMM YYYY"
      navNext={
        <span
          className="icon-chevron-right"
        />
      }
      navPrev={
        <span
          className="icon-chevron-left"
        />
      }
      numberOfMonths={1}
      onBlur={[Function]}
      onClose={[Function]}
      onDatesChange={[Function]}
      onFocusChange={[Function]}
      onNextMonthClick={[Function]}
      onOutsideClick={[Function]}
      onPrevMonthClick={[Function]}
      orientation="horizontal"
      phrases={
        Object {
          "calendarLabel": "Calendar",
          "chooseAvailableDate": [Function],
          "chooseAvailableEndDate": [Function],
          "chooseAvailableStartDate": [Function],
          "dateIsUnavailable": [Function],
          "enterKey": "Enter key",
          "escape": "Escape key",
          "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
          "homeEnd": "Home and end keys",
          "jumpToNextMonth": "Move forward to switch to the next month",
          "jumpToPrevMonth": "Move backward to switch to the previous month",
          "keyboardShortcuts": "Keyboard Shortcuts",
          "leftArrowRightArrow": "Right and left arrow keys",
          "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
          "moveFocusByOneMonth": "Switch months",
          "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
          "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
          "openThisPanel": "Open this panel",
          "pageUpPageDown": "page up and page down keys",
          "questionMark": "Question mark",
          "returnFocusToInput": "Return to the date input field",
          "selectFocusedDate": "Select the date in focus",
          "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
          "upArrowDownArrow": "up and down arrow keys",
        }
      }
      renderCalendarInfo={null}
      renderDay={null}
      renderMonth={null}
      showKeyboardShortcuts={false}
      startDate={"2018-01-01T00:00:00.000Z"}
      verticalHeight={null}
      weekDayFormat="dd"
      withPortal={false}
    />
  </div>
</div>
`;
